src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,152 @@
+                }
+            } catch (Exception e) {
+                handleExceptions(e);
+            }
+        }
+        handleExitProgram();
+    }
+
+    public static void printAddedTask(ArrayList<Task> list) {
+        System.out.println("____________________________________________________________");
+        System.out.println("    Got it. I've added this task:");
+        System.out.println("    " + list.get(list.size()-1).getLabel());
+        System.out.println("    Now you have " + list.size() + " tasks in the list.");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleShowTaskList(ArrayList<Task> list) {
+        System.out.println("____________________________________________________________");
+        System.out.println("    Here are the tasks in your list:");
+        for (int i = 0; i < list.size(); i++) {
+            String prefix = (i + 1) + ".";
+            System.out.println("    " + prefix + list.get(i).getLabel());
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleDeleteTask(ArrayList<Task> list, int index) {
+        list.remove(index-1);
+        System.out.println("____________________________________________________________");
+        System.out.println("    Noted. I've removed this task:");
+        System.out.println(list.get(index-1).getLabel());
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleMarkTask(ArrayList<Task> list, int indexToMark) {
+        list.get(indexToMark).setIsDone(true);
+        System.out.println("____________________________________________________________");
+        System.out.println("    Nice! I've marked this task as done:");
+        System.out.println(list.get(indexToMark).getLabel());
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleUnmarkTask(ArrayList<Task> list, int indexToUnmark) {
+        list.get(indexToUnmark).setIsDone(false);
+        System.out.println("____________________________________________________________");
+        System.out.println("    OK, I've marked this task as not done yet:");
+        System.out.println(list.get(indexToUnmark).getLabel());
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleExitProgram() {
+        System.out.println("____________________________________________________________");
+        System.out.println("    Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void handleExceptions(Exception e) {
+        switch (e.getClass().getName()) {
+            case "duke.IllegalCommandException":
```

Comment:
Do look out for the correct indentation for switch-cases. There should not be 4 identation spaces for the `case` lines. Take a look at the Java coding standard document if you are unsure on this.

Reply:
Also, a more common way of handling the exceptions is to handle it within the `catch` block within a `try-catch`.